test/core/V3/GenArt721CoreV3_ProjectConfigure.test.ts

```diff
@@ -714,6 +727,173 @@ describe("GenArt721CoreV3 Project Configure", async function () {
     });
   });
 
+  describe("addProjectScript", function () {
+    it("uploads and recalls an empty script", async function () {
+      await this.genArt721Core
+        .connect(this.accounts.artist)
+        .addProjectScript(this.projectZero, "");
+      const script = await this.genArt721Core.projectScriptByIndex(
+        this.projectZero,
+        0
+      );
+      expect(script).to.equal("");
+    });
+
+    it("uploads and recalls an short script < 32 bytes", async function () {
+      const targetScript = "console.log(hello world)";
+      await this.genArt721Core
+        .connect(this.accounts.artist)
+        .addProjectScript(this.projectZero, targetScript);
+      const script = await this.genArt721Core.projectScriptByIndex(
+        this.projectZero,
+        0
+      );
+      expect(script).to.equal(targetScript);
+    });
+
+    it("uploads and recalls chromie squiggle script", async function () {
+      await this.genArt721Core
+        .connect(this.accounts.artist)
+        .addProjectScript(this.projectZero, SQUIGGLE_SCRIPT);
+      const script = await this.genArt721Core.projectScriptByIndex(
+        this.projectZero,
+        0
+      );
+      expect(script).to.equal(SQUIGGLE_SCRIPT);
+    });
+
+    it("uploads chromie squiggle script, and then purges it", async function () {
+      await this.genArt721Core
+        .connect(this.accounts.artist)
+        .addProjectScript(this.projectZero, SQUIGGLE_SCRIPT);
+      const script = await this.genArt721Core.projectScriptByIndex(
+        this.projectZero,
+        0
+      );
+      expect(script).to.equal(SQUIGGLE_SCRIPT);
+
+      const scriptAddress =
+        await this.genArt721Core.projectScriptBytecodeAddressByIndex(
+          this.projectZero,
+          0
+        );
+
+      const scriptByteCode = await ethers.provider.getCode(scriptAddress);
+      expect(scriptByteCode).to.not.equal("0x");
+
+      await this.genArt721Core
+        .connect(this.accounts.artist)
+        .removeProjectLastScript(this.projectZero);
+      await expectRevert(
+        this.genArt721Core.projectScriptByIndex(this.projectZero, 0),
+        "ContractAsStorage: Read Error"
+      );
+
+      const removedScriptByteCode = await ethers.provider.getCode(
+        scriptAddress
+      );
+      expect(removedScriptByteCode).to.equal("0x");
+    });
+
+    it("uploads chromie squiggle script and attempts to purge from non-allowed address", async function () {
```

Comment:
Thinking on this - we should add a similar test to this that specifically contains a bunch of valid utf-8 characters >1 byte

Reply:
Updated to add a specific test case for this that uses a newly introduced `RANDOM_UTF_EIGHT_SCRIPT` mocked up script with a bunch of utf-8 characters.